src/core/cache.jl

```diff
@@ -0,0 +1,25 @@
+struct CacheKey{C <: AbstractCache, D <: PSY.Device}
+    cache_type::Type{C}
+    device_type::Type{D}
+end
+
+function CacheKey(cache::C) where {C <: AbstractCache}
+    return CacheKey(C, cache.device_type)
+end
+
+"""
+Tracks the last time status of a device changed in a simulation
+"""
+# The current implementation will require all custom caches to have the same data template
+mutable struct TimeStatusChange <: AbstractCache
+    device_type::Type{<:PSY.Device}
+    value::JuMP.Containers.DenseAxisArray{Dict{Symbol, Float64}}
+    ref::UpdateRef
+end
```

Comment:
The docstring for `TimeStatusChange` will not show up.

<img width="890" alt="Screen Shot 2020-02-19 at 1 45 40 PM" src="https://user-images.githubusercontent.com/1813121/74874392-20a48a80-531e-11ea-9928-34ee24594b52.png">